pkg/harvester-manager/machine-config/harvester.vue

```diff
@@ -909,14 +933,14 @@ export default {
     },
 
     updateVGpu() {
-      const vGPURequests = this.vGpus?.filter((name) => name).reduce((acc, name, i) => ([
-        ...acc,
-        {
-          name,
-          deviceName: this.vGpuEnabledDevices[name]?.type,
-        }
-      ])
-      , []);
+      /**
+       * We are assigning the first vGpu profile found for each vGpu type selected by the user.
+       * This will not work if we will remove the limit of only one vGpu assignable to each cluster.
+       */
+      const vGPURequests = this.vGpus?.filter((f) => f).map((deviceName) => ({
+        name: Object.values(this.vGpuDevices).filter((f) => f.type === deviceName)?.[0]?.id || '',
```

Comment:
Hi @torchiaf, are we keep assigning first vGPU profile id as payload name here ? 

I know we had a concern about it before. 

Since [there](https://github.com/harvester/pcidevices/pull/91) introduces a new annotation `harvesterhci.io/deviceAllocationDetails` in VM. 

We need another PR in harvester dashboard to leverage that annotation.

Reply:
Correct, in Rancher we can pass a default name (better if it's something explaining the root device); on Harvester we will need to get the device id from the label rather than the vm's spec.